src/Core/MySQL/PacketsProtocolText.cpp

```diff
@@ -131,16 +134,25 @@ void ColumnDefinition::writePayloadImpl(WriteBuffer & buffer) const
     }
 }
 
-ColumnDefinition getColumnDefinition(const String & column_name, const TypeIndex type_index)
+ColumnDefinition getColumnDefinition(const String & column_name, const DataTypePtr & data_type)
 {
     ColumnType column_type;
     CharacterSet charset = CharacterSet::binary;
     int flags = 0;
+    uint8_t decimals = 0;
+    TypeIndex type_index = data_type->getTypeId();
     switch (type_index)
     {
         case TypeIndex::UInt8:
-            column_type = ColumnType::MYSQL_TYPE_TINY;
-            flags = ColumnDefinitionFlags::BINARY_FLAG | ColumnDefinitionFlags::UNSIGNED_FLAG;
+            if (data_type->getName() == "Bool")
+            {
+                column_type = ColumnType::MYSQL_TYPE_BIT;
```

Comment:
Just curious: MySQL has the `BIT` type only since v8.0. Before that, one had to use `BOOL`/`BOOLEAN`. The latter are aliases of TINYINT, i.e. 1 byte integers. That's also what "Bool" is in ClickHouse. But what is `BIT` physically in MySQL? I had some trouble finding out. If it is a 64-bit value (and it seems at least so), then `BOOL`/`BOOLEAN` would probably better fits.

Reply:
I think we can use TINYINT for that as well. I will check.

Reply:
It should work at least since MySQL 5.7: https://dev.mysql.com/doc/refman/5.7/en/bit-type.html

SO suggests it is this way since at least 2008: https://stackoverflow.com/a/290253/4575540